sjsonnet/src/sjsonnet/Std.scala

```diff
@@ -213,113 +213,113 @@ class Std(private val additionalNativeFunctions: Map[String, Val.Builtin] = Map.
   }
 
   private object Any extends Val.Builtin1("any", "arr") {
-    def evalRhs(arr: Val, ev: EvalScope, pos: Position): Val = {
-      Val.bool(pos, arr.asArr.iterator.exists(v => v.asBoolean))
+    def evalRhs(arr: Lazy, ev: EvalScope, pos: Position): Val = {
+      Val.bool(pos, arr.force.asArr.iterator.exists(v => v.asBoolean))
     }
   }
 
   private object Type extends Val.Builtin1("type", "x") {
-    def evalRhs(x: Val, ev: EvalScope, pos: Position): Val = Val.Str(pos, x.prettyName)
+    def evalRhs(x: Lazy, ev: EvalScope, pos: Position): Val = Val.Str(pos, x.force.prettyName)
   }
 
   private object Format_ extends Val.Builtin2("format", "str", "vals") {
-    def evalRhs(str: Val, vals: Val, ev: EvalScope, pos: Position): Val =
-      new Val.Str(pos, Format.format(str.asString, vals, pos)(ev))
-    override def specialize(args: Array[Expr]) = args match {
+    def evalRhs(str: Lazy, vals: Lazy, ev: EvalScope, pos: Position): Val =
+      new Val.Str(pos, Format.format(str.force.asString, vals.force, pos)(ev))
+    override def specialize(args: Array[Expr], tailstrict: Boolean) = args match {
       case Array(str, fmt: Val.Str) =>
         try { (new Format.PartialApplyFmt(fmt.value), Array(str)) } catch { case _: Exception => null }
       case _ => null
     }
   }
 
   private object Foldl extends Val.Builtin3("foldl", "func", "arr", "init") {
-    def evalRhs(_func: Val, arr: Val, init: Val, ev: EvalScope, pos: Position): Val = {
-      val func = _func.asFunc
-      arr match{
+    def evalRhs(_func: Lazy, arr: Lazy, init: Lazy, ev: EvalScope, pos: Position): Val = {
+      val func = _func.force.asFunc
+      arr.force match {
         case arr: Val.Arr =>
-          var current = init
+          var current = init.force
           for (item <- arr.asLazyArray) {
             val c = current
             current = func.apply2(c, item, pos.noOffset)(ev)
           }
           current
 
         case s: Val.Str =>
-          var current = init
+          var current = init.force
           for (char <- s.value) {
             val c = current
             current = func.apply2(c, Val.Str(pos, new String(Array(char))), pos.noOffset)(ev)
           }
           current
 
-        case _ => Error.fail("Cannot call foldl on " + arr.prettyName)
+        case arr => Error.fail("Cannot call foldl on " + arr.prettyName)
       }
 
 
     }
   }
 
   private object Foldr extends Val.Builtin3("foldr", "func", "arr", "init") {
-    def evalRhs(_func: Val, arr: Val, init: Val, ev: EvalScope, pos: Position): Val = {
-      val func = _func.asFunc
+    def evalRhs(_func: Lazy, arr: Lazy, init: Lazy, ev: EvalScope, pos: Position): Val = {
+      val func = _func.force.asFunc
       arr match {
```

Comment:
Should this be `arr.force` here? I see that you have the `arr.force` below in the log line instead, but I think that should be swapped for symmetry with other usage.